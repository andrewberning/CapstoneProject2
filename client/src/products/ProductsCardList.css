.products-list {
  display: flex;
  flex-flow: wrap;
  justify-content: space-evenly;
  margin-bottom: 0;
  padding-left: 0;
  list-style: none;
}