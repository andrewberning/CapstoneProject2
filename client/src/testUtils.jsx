import React from "react";
import UserContext from "./auth/UserContext";
import { vi } from 'vitest';

const demoUser = {
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "testuser123@test.net",
};

// Mock functions
const mockSetCurrentUser = vi.fn();
const mockLogin = vi.fn();
const mockLogout = vi.fn();
const mockSignup = vi.fn();
const mockAddToCart = vi.fn();
const mockHasAddedToCart = vi.fn();

const UserProvider = ({ 
  children,
  currentUser = demoUser,
  setCurrentUser = mockSetCurrentUser,
  login = mockLogin,
  logout = mockLogout,
  signup = mockSignup,
  addToCart = mockAddToCart,
  hasAddedToCart = mockHasAddedToCart,
  cartItems = [],
  setCartItems = () => {},
  categories = [],
  infoLoaded = true,
  totalAmount = 0,
  totalItems = 0,
}) => (
  <UserContext.Provider value={{ 
    currentUser,
    setCurrentUser,
    login,
    logout,
    signup,
    addToCart,
    hasAddedToCart,
    cartItems,
    setCartItems,
    categories,
    infoLoaded,
    totalAmount,
    totalItems,
  }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider, mockSetCurrentUser, mockLogin, mockLogout, mockSignup, mockAddToCart, mockHasAddedToCart };
