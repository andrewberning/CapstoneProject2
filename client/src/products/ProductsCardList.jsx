import { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import ShoplyApi from "../api/api";
import ProductCard from "./ProductCard";
import "./ProductsCardList.css";

export default function ProductsList() {
  const { category } = useParams();

  const [products, setProducts] = useState([])

  useEffect(function getCategoryProductsOnMount() {
    async function getCategoryProducts() {
      if (!category) return;
      let categoryData = await ShoplyApi.getCategory(category);
      let categoryId = categoryData.category.id;
      let products = await ShoplyApi.getProductsByCategoryId(categoryId);
      setProducts(products);
    }

    getCategoryProducts();
  }, [category])

  if (!category) {
    return <p className="lead text-center">Category not found.</p>
  }

  return (
    <div className="ProductsCardList">
      <h2 className="text-center">{category.charAt(0).toUpperCase() + category.slice(1)}</h2>
      {products.length
        ? (
            <ul className="products-list">
              {products.map(product => (
                <ProductCard 
                key={product.id}
                id={product.id}
                category={category}
                name={product.name}
                description={product.description}
                price={product.price}
                image={product.image_url}
                />
              ))}
            </ul>
        ) : (
          <p className="lead text-center">Sorry, no results were found.</p>
        )
      }

    </div>
  );
}