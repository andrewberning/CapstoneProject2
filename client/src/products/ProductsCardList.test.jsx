import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import ProductsCardList from "./ProductsCardList";
import { UserProvider } from "../testUtils";


describe('ProductsCardList component', () => {

  it('renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserProvider >
          <ProductsCardList />
        </UserProvider>
      </MemoryRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <ProductsCardList />
        </UserProvider>
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
})