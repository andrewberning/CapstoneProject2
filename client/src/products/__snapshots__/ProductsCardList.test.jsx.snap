// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductsCardList component > matches snapshot 1`] = `
<DocumentFragment>
  <p
    class="lead text-center"
  >
    Category not found.
  </p>
</DocumentFragment>
`;
